packages/suite/src/components/suite/SecurityCheck/SecurityCheckFail.tsx

```diff
@@ -59,44 +40,35 @@ const checklistItems = [
     },
 ] as const;
 
-const softChecklistItems = [
-    {
-        icon: 'plugs',
-        content: <Translation id="TR_DISCONNECT_DEVICE_SOFT" />,
-    },
-    {
-        icon: 'hand',
-        content: <Translation id="TR_AVOID_USING_DEVICE_SOFT" />,
-    },
-    {
-        icon: 'chat',
-        content: <Translation id="TR_USE_CHAT_SOFT" values={{ b: chunks => <b>{chunks}</b> }} />,
-    },
-] as const;
-
 const supportChatUrl = `${TREZOR_SUPPORT_URL}#open-chat`;
 
 interface SecurityCheckFailProps {
     goBack?: () => void;
+    useSoftMessaging?: boolean;
 }
 
-export const SecurityCheckFail = ({ goBack }: SecurityCheckFailProps) => {
-    const heading = goBack ? 'TR_DEVICE_COMPROMISED_HEADING_SOFT' : 'TR_DEVICE_COMPROMISED_HEADING';
-    const text = goBack ? 'TR_DEVICE_COMPROMISED_TEXT_SOFT' : 'TR_DEVICE_COMPROMISED_TEXT';
-    const items = goBack ? softChecklistItems : checklistItems;
+export const SecurityCheckFail = ({ goBack, useSoftMessaging }: SecurityCheckFailProps) => {
+    const heading = useSoftMessaging
```

Comment:
I don't know about this, it feels too pedantic to me.

Ad 1, it is questionable if this saves you time because you have to type more.
Ad 2, I agree.
3) This breaks the inference as `heading` becomes a general `TranslationKey` instead of a literal union.

It could be a topic for Code Improvements though.

Reply:
:thinking:  :thought_balloon:  **If** inference isn't enough, and we want explicit declaration for stricter typesafety _(imho we don't)_, then how about `satisfies`?
It'd tell us if `useSoftMessaging` returns something else than we expect, but at the same time, keep type inferrence, which is more specific than `TranslationKey`